pennylane_lightning/core/src/simulators/lightning_kokkos/measurements/MeasurementsKokkos.hpp

```diff
@@ -509,117 +513,68 @@ class Measurements final
      */
     auto probs(const std::vector<std::size_t> &wires)
         -> std::vector<PrecisionT> {
-        PL_ABORT_IF_NOT(
-            std::is_sorted(wires.cbegin(), wires.cend()),
-            "LightningKokkos does not currently support out-of-order wire "
-            "indices with probability calculations");
-
-        // If all wires are requested, dispatch to `this->probs()`
-        if (wires.size() == this->_statevector.getNumQubits()) {
-            return this->probs();
+        const std::size_t n_wires = wires.size();
+        if (n_wires == 0) {
+            return {1.0};
         }
-
-        using MDPolicyType_2D =
-            Kokkos::MDRangePolicy<Kokkos::Rank<2, Kokkos::Iterate::Left>>;
-
-        //  Determining probabilities for the sorted wires.
-        const Kokkos::View<ComplexT *> arr_data = this->_statevector.getView();
         const std::size_t num_qubits = this->_statevector.getNumQubits();
-
-        std::vector<std::size_t> sorted_ind_wires(wires);
-        const bool is_sorted_wires =
-            std::is_sorted(sorted_ind_wires.begin(), sorted_ind_wires.end());
-        std::vector<std::size_t> sorted_wires(wires);
-
-        if (!is_sorted_wires) {
-            sorted_ind_wires = Pennylane::Util::sorting_indices(wires);
-            for (std::size_t pos = 0; pos < wires.size(); pos++)
-                sorted_wires[pos] = wires[sorted_ind_wires[pos]];
+        bool is_equal_to_all_wires = n_wires == num_qubits;
+        for (std::size_t k = 0; k < n_wires; k++) {
+            if (!is_equal_to_all_wires) {
+                break;
+            }
+            is_equal_to_all_wires = wires[k] == k;
+        }
+        if (is_equal_to_all_wires) {
+            return this->probs();
+        }
+        const bool is_gpu_scratch_limited =
+            n_wires > 7 && !std::is_same_v<KokkosExecSpace, HostExecSpace>;
```

Comment:
For `n_wires > 7`, Kokkos will complain that (on A100 cards) it does not have enough L0 scratch memory.